kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/adapter/SQLConnection.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.jdbc.hive.adapter;
+
+import java.sql.*;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Executor;
+
+public interface SQLConnection extends Connection {
+
+  @Override
+  default Statement createStatement() throws SQLException {
+    throw new SQLFeatureNotSupportedException("Method not supported");
+  }
+
+  @Override
+  default PreparedStatement prepareStatement(String sql) throws SQLException {
+    throw new SQLFeatureNotSupportedException("Method not supported");
+  }
+
+  @Override
+  default CallableStatement prepareCall(String sql) throws SQLException {
+    throw new SQLFeatureNotSupportedException("Method not supported");
+  }
+
+  @Override
+  default String nativeSQL(String sql) throws SQLException {
+    throw new SQLFeatureNotSupportedException("Method not supported");
+  }
+
+  @Override
+  default void setAutoCommit(boolean autoCommit) throws SQLException {
+    throw new SQLFeatureNotSupportedException("Method not supported");
+  }
+
+  @Override
+  default boolean getAutoCommit() throws SQLException {
+    throw new SQLFeatureNotSupportedException("Method not supported");
+  }
+
+  @Override
+  default void commit() throws SQLException {
+    throw new SQLFeatureNotSupportedException("Method not supported");
+  }
+
+  @Override
+  default void rollback() throws SQLException {
+    throw new SQLFeatureNotSupportedException("Method not supported");
+  }
+
+  @Override
+  default void close() throws SQLException {
```

Comment:
We can remove the default method in adaptor layer for those implemented methods.